test/irmin-pack/concurrent.ml

```diff
@@ -0,0 +1,191 @@
+open Lwt.Infix
+
+let root = Filename.concat "_build" "test-concurrent"
+
+let src = Logs.Src.create "tests.unix.concurrent" ~doc:"Tests"
+
+module Log = (val Logs.src_log src : Logs.LOG)
+
+let index_log_size = Some 1_000
+
+let parent_pid = Unix.getpid ()
+
+module Conf = struct
+  let entries = 32
+
+  let stable_hash = 256
+
+  let keep_max = true
+end
+
+module Hash = Irmin.Hash.SHA1
+module S =
+  Irmin_pack.Make (Conf) (Irmin.Metadata.None) (Irmin.Contents.String)
+    (Irmin.Path.String_list)
+    (Irmin.Branch.String)
+    (Hash)
+
+let config ?(readonly = false) ?(fresh = true) root =
+  Irmin_pack.config ~readonly ?index_log_size ~fresh root
+
+let info () = Irmin.Info.empty
+
+let init () =
+  if Sys.file_exists root then (
+    let cmd = Printf.sprintf "rm -rf %s" root in
+    Fmt.epr "exec: %s\n%!" cmd;
+    let _ = Sys.command cmd in
+    () )
+
+(* Helpers for Unix pipe *)
+let write input =
+  try
+    let m = Bytes.of_string (string_of_int (Unix.getpid ())) in
+    let n = Unix.write input m 0 (Bytes.length m) in
+    assert (n = Bytes.length m)
+  with Unix.Unix_error (n, f, arg) ->
+    Alcotest.failf "Unix error %s %s %s " f arg (Unix.error_message n)
```

Comment:
Does Alcotest not print such exceptions by default?

Reply:
it does :)